addons/mail/models/mail_thread.py

```diff
@@ -283,10 +283,15 @@ def create(self, vals_list):
             threads_no_subtype = self.env[self._name]
             for thread in threads:
                 subtype = thread._creation_subtype()
-                if subtype:  # if we have a subtype, post message to notify users from _message_auto_subscribe
-                    thread.sudo().message_post(subtype_id=subtype.id, author_id=self.env.user.partner_id.id)
-                else:
+                if not subtype:
                     threads_no_subtype += thread
+                    continue
+                # if we have a subtype, post message to notify users from _message_auto_subscribe
+                thread.sudo().message_post(
+                    subtype_id=subtype.id, author_id=self.env.user.partner_id.id,
+                    # summary="o_mail_notification" is used to hide the message body in the front-end
+                    body=Markup('<div summary="o_mail_notification"><p>%s</p></div>') % thread._creation_message()
```

Comment:
I see that some overrides of _creation_message actually contain html (see in hr expense notably). Shouldn't ``creation_message`` return Markup content so that either we give raw html and it is escaped, either we give markup and we keep it here ?